lnwire/stfu.go

```diff
@@ -0,0 +1,72 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+)
+
+// Stfu is a message that is sent to lock the channel state prior to some other
+// interactive protocol where channel updates need to be paused.
+type Stfu struct {
+	// ChanID identifies which channel needs to be frozen.
+	ChanID ChannelID
+
+	// Initiator is a byte that identifies whether we are the initiator of
+	// this process.
+	Initiator bool
+
+	// ExtraData is the set of data that was appended to this message to
+	// fill out the full maximum transport message size. These fields can
+	// be used to specify optional data such as custom TLV fields.
+	ExtraData ExtraOpaqueData
+}
+
+// A compile time check to ensure Stfu implements the lnwire.Message interface.
+var _ Message = (*Stfu)(nil)
+
+// Encode serializes the target Stfu into the passed io.Writer.
+// Serialization will observe the rules defined by the passed protocol version.
+//
+// This is a part of the lnwire.Message interface.
+func (s *Stfu) Encode(w *bytes.Buffer, _ uint32) error {
+	var extraBytesWriter bytes.Buffer
+	s.ExtraData = ExtraOpaqueData(extraBytesWriter.Bytes())
```

Comment:
Unnecessary - we can just have `WriteBytes` below?

If we get to the stage where we're actually encoding TLVs here then we could use `EncodeMessageExtraData`, not an empty buffer.

Reply:
This is the stuff I don't have a deep knowledge of the nuance of. I'll admit I made it this way because I copied other stuff I saw. I'll see what can be done.

Reply:
I remember why I did things this way. It's because it was the first way I could figure out to make the fuzz tests that check for round trip encode/decode equality pass. I have tried something different/new that passes and hopefully has fewer loose ends.